sqlx-postgres/src/listener.rs

```diff
@@ -69,6 +71,7 @@ impl PgListener {
             buffer_rx: receiver,
             buffer_tx: None,
             channels: Vec::new(),
+            span: Span::current(),
```

Comment:
Capturing `Span::current()` on construction of the listener is potentially controversial, I think.

We can't assume how people are making use of `tracing`. This may end up being undesirable behavior.

If we did support something like this, I would probably expect it to be an explicit setter.

Reply:
Thank you for looking into this.

I understand that this approach is controversial, but the current situation does not seem to be desirable to some extent as well. What about using the `Span` that defaults to capture `Span::current()` on construction as is and adding a setter as you suggested? If that is an option, I would also be curious as to how that setter is named.

Reply:
It's probably better to just capture the current span on-drop. That's arguably more useful anyway if you actually want to know _where_ it's being dropped.

Reply:
As I tested, there was no any span available with `Span::current()` in the drop handler.

Reply:
There should be a span available if the outer future or async block is properly instrumented: https://docs.rs/tracing/latest/src/tracing/instrument.rs.html#277

Reply:
You are right that the current span is available in the drop handler, but only when `into_stream()` is not involved; it turned out that the root cause is the `try_stream` macro that internally `move`s this `PgListener` to a `Future`, hence lost the span. I updated the PR to reflect the change, so could you look into this again?

Reply:
Looks good to me!